core/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -367,59 +364,65 @@ public void testNamedWriteableList() throws IOException {
     }
 
     public void testNamedWriteableNotSupportedWithoutWrapping() throws IOException {
-        BytesStreamOutput out = new BytesStreamOutput();
-        TestNamedWriteable testNamedWriteable = new TestNamedWriteable("test1", "test2");
-        out.writeNamedWriteable(testNamedWriteable);
-        StreamInput in = StreamInput.wrap(BytesReference.toBytes(out.bytes()));
-        try {
-            in.readNamedWriteable(BaseNamedWriteable.class);
-            fail("Expected UnsupportedOperationException");
-        } catch (UnsupportedOperationException e) {
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            ;
+            TestNamedWriteable testNamedWriteable = new TestNamedWriteable("test1", "test2");
+            out.writeNamedWriteable(testNamedWriteable);
+            StreamInput in = StreamInput.wrap(BytesReference.toBytes(out.bytes()));
+            Exception e = expectThrows(UnsupportedOperationException.class, () -> in.readNamedWriteable(BaseNamedWriteable.class));
             assertThat(e.getMessage(), is("can't read named writeable from StreamInput"));
         }
     }
 
     public void testNamedWriteableReaderReturnsNull() throws IOException {
-        BytesStreamOutput out = new BytesStreamOutput();
-        NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry(Collections.singletonList(
-            new NamedWriteableRegistry.Entry(BaseNamedWriteable.class, TestNamedWriteable.NAME, (StreamInput in) -> null)
-        ));
-        TestNamedWriteable namedWriteableIn = new TestNamedWriteable(randomAsciiOfLengthBetween(1, 10), randomAsciiOfLengthBetween(1, 10));
-        out.writeNamedWriteable(namedWriteableIn);
-        byte[] bytes = BytesReference.toBytes(out.bytes());
-        StreamInput in = new NamedWriteableAwareStreamInput(StreamInput.wrap(bytes), namedWriteableRegistry);
-        assertEquals(in.available(), bytes.length);
-        IOException e = expectThrows(IOException.class, () -> in.readNamedWriteable(BaseNamedWriteable.class));
-        assertThat(e.getMessage(), endsWith("] returned null which is not allowed and probably means it screwed up the stream."));
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            ;
+            NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry(Collections.singletonList(
+                    new NamedWriteableRegistry.Entry(BaseNamedWriteable.class, TestNamedWriteable.NAME, (StreamInput in) -> null)));
+            TestNamedWriteable namedWriteableIn = new TestNamedWriteable(randomAsciiOfLengthBetween(1, 10),
+                    randomAsciiOfLengthBetween(1, 10));
+            out.writeNamedWriteable(namedWriteableIn);
+            byte[] bytes = BytesReference.toBytes(out.bytes());
+            try (StreamInput in = new NamedWriteableAwareStreamInput(StreamInput.wrap(bytes), namedWriteableRegistry)) {
+                ;
```

Comment:
And one more...they seem to be all over, I presume a mistake in regex find/replace.